### PR TITLE
[Serve][Doc] Fix pydantic config documentation

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

Current (master branch) schema API ref:
https://docs.ray.io/en/master/serve/api/doc/ray.serve.schema.ServeDeploySchema.html

This change:
https://anyscale-ray--42216.com.readthedocs.build/en/42216/serve/api/doc/ray.serve.schema.ServeDeploySchema.html#ray.serve.schema.ServeDeploySchema

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/requirements-doc.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ sphinxcontrib-redoc==1.6.0
 sphinx-remove-toctrees==0.0.3
 sphinx_design==0.5.0
 pydata-sphinx-theme==0.14.1
-autodoc_pydantic==2.0.1
+autodoc_pydantic==1.9.0
```
Comment (sihanwang41) [inline doc/requirements-doc.txt:19]: autodoc_pydantic 2 is not working with pydantic v1.

File: doc/source/serve/api/index.md [hunk 2]
```diff
@@ -355,7 +355,6 @@ Content-Type: application/json
 .. autosummary::
    :nosignatures:
    :toctree: doc/
-   :template: autosummary/class_with_inherited_members.rst
```
Comment (edoakes) [inline doc/source/serve/api/index.md:358]: should this be deleted then?

Comment (edoakes) [reply doc/source/serve/api/index.md:358]: I think it was only added for this purpose

**Review by edoakes — APPROVED** [review]

**Review by angelinalg — APPROVED** [review]

File: doc/source/serve/api/index.md (hunk 2 above)
Comment (sihanwang41) [reply doc/source/serve/api/index.md:358]: oh, good catch, removed.

**Comment by sihanwang41** [comment]
@edoakes ping to merge.

**Pull request merged**